pkg/config/config.go

```diff
@@ -704,10 +703,26 @@ func (conf *Config) updateFromCLI(c *cli.Context, baseFlags []cli.Flag) error {
 		case reflect.Int, reflect.Int32, reflect.Int64:
 			configValue.SetInt(c.Int64(flagName))
 		case reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
-			configValue.SetUint(c.Uint64(flagName))
+			value := c.Value(flagName)
+			switch value.(type) {
+			case uint:
+				configValue.SetUint(uint64(value.(uint)))
+			case uint8:
+				configValue.SetUint(uint64(value.(uint8)))
+			case uint16:
+				configValue.SetUint(uint64(value.(uint16)))
+			case uint32:
+				configValue.SetUint(uint64(value.(uint32)))
+			case uint64:
+				configValue.SetUint(uint64(value.(uint64)))
+			default:
+				return fmt.Errorf("unsupported generated cli flag type for config: %s is a %s", flagName, kind.String())
```

Comment:
If I try to do `c.Uint64()` here, it just gives me `0`.

Missed doing this for signed int, let me do it, please check if any other cases need this special handling

Reply:
looks like we'll need to cast to the target type
```go
configValue.Set(reflect.ValueOf(c.Value(flagName)).Convert(configValue.Type()))
```

Reply:
actually you can replace the whole type switch with
```go
value := reflect.ValueOf(c.Value(flagName))
if value.CanConvert(configValue.Type()) {
	configValue.Set(value.Convert(configValue.Type()))
} else {
	return fmt.Errorf("unsupported generated cli flag type for config: %s is a %s", flagName, kind.String())
}
```

Reply:
you could include the config/value types in the error message to make it more helpful - eg "wrong type for x expected y got z"

Reply:
This is great, thanks!

Reply:
re-requested a review just to make sure everything is alright after the new changes